libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -5150,6 +5153,48 @@ void GCS_MAVLINK::send_uavionix_adsb_out_status() const
 #endif
 }
 
+void GCS_MAVLINK::send_autopilot_state_for_gimbal_device() const
+{
+    // get attitude
+    const AP_AHRS &ahrs = AP::ahrs();
+    Quaternion quat;
+    if (!ahrs.get_quaternion(quat)) {
+        return;
+    }
+    const float repr_offseq_q[] = {quat.q1, quat.q2, quat.q3, quat.q4};
+
+    // get velocity
+    Vector3f vel;
+    if (!ahrs.get_velocity_NED(vel)) {
+        vel.zero();
+    }
+
+    // get body-frame rotation rates
+    const Vector3f& omega = ahrs.get_gyro();
+
+    // get estimator flags
+    uint16_t est_status_flags = 0;
+    nav_filter_status nav_filt_status;
+    if (ahrs.get_filter_status(nav_filt_status)) {
+        est_status_flags = (uint16_t)(nav_filt_status.value & 0xFFFF);
+    }
+
+    mavlink_msg_autopilot_state_for_gimbal_device_send(
+        chan,
+        mavlink_system.sysid,   // target system (this autopilot's gimbal)
+        0,                  // target component (anything)
+        AP_HAL::micros(),   // time boot us
+        repr_offseq_q,  // attitude as quaternion
+        0,      // attitude estimated delay in micros
+        vel.x,  // x speed in NED (m/s)
+        vel.y,  // y speed in NED (m/s)
+        vel.z,  // z speed in NED (m/s)
+        0,      // velocity estimated delay in micros
+        omega.z,// feed forward angular velocity z
```

Comment:
no, the docs are not extremely clear, but to the best of my knowledge and understanding and common sense this should not be the vehicles current yaw vel, but the intended yaw vel, i.e., the value which the vehicle is commanded to do (i.e., the setpoint in controller language)(and feedforward like feedforward in control language).
Otherwise the gimbal would have to do some filtering and fancy guessing to know what's going on.

Reply:
yes, you must be right.  This should be the vehicle's target yaw rate.. I'll fix this.

Reply:
done.